### ISSUE TITLE
[Bug]: flatpak-builder skips many modules if 1 doesn't exist.

### ISSUE THREAD

**Issue by BigmenPixel0** [flatpak/flatpak-builder#471]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak-builder/issues) for a bug that matches the one I want to file, without success.

### flatpak-builder version

1.2.2

### Flatpak version

1.12.7

### How to reproduce

1. Write a manifest with `modules:` in a module. Example:
```
- name: module1
  ...
  modules:
    - shared-modules/libusb/libusb.json
    
    - name: module2
      ...
 ```
2. `shared-modules` directory does not exist.
3. `flatpak-builder` skips what's in `modules:` and builds `module1`.

### Expected Behavior

Skip `libusb` module.

### Actual Behavior

```
(flatpak-builder:2): Json-WARNING **: 10:20:09.021: Failed to deserialize "modules" property of type "gpointer" for an object of type "BuilderModule"
```
And the build of `module1` began. (`module2` skipped)

**Comment by TingPing** [comment]
It should probably just fatally error. This was partially fixed in #424 but not for deserialize errors.

**Comment by gasinvein** [comment]
This one would be a bit trickier to fix, sine, to my understanding, the warning is emitted from json-glib and not directly from flatpak-builder. Can we somehow tell json-glib to hard fail on any parse error?
